test/expect-tests/dune_memory/memory.ml

```diff
@@ -86,15 +86,20 @@ let%expect_test _ =
     >>| fun searched ->
     ( match searched with
     | stored_metadata, [ file ] ->
+      let in_the_build_directory =
+        Path.of_local (Path.Build.local file.in_the_build_directory)
```

Comment:
This conversion is very strange. I think it results in a `In_source_dir` constructor of `Path.t`, which makes no sense in this context.

Reply:
How do I use my `Path.Local.t` with the many functions from `stdune` that take a `Path.t` ? E.g. `Digest.file_with_stats`, `Io.compare_files`, `Path.equal` ...

Reply:
Ah, I didn't think about that. I guess those functions were never intended for use outside of dune so things don't really make sense. 
It might be worth leaving a comment explaining the need for this.

In the long run, I think we should have a filesystem access layer that doesn't make assumptions about being a part of dune.